### PR TITLE
feat: add a proper error message when file is too big

### PR DESCRIPTION
**Pull request by Pierre-Monier**
[#1257](https://github.com/excalidraw/excalidraw/issues/1257)
[old overkill merge request](https://github.com/excalidraw/excalidraw/pull/2231)

feat: check if error_class in server reponse is RequestTooLargeError and display a proper error message. 

P-S: In dev env the server isn't responding with error_class -> RequestTooLargeError

### CODE REVIEW

File: src/data/index.ts [hunk 1]
```diff
@@ -218,13 +218,13 @@ export const exportToBackend = async (
...
       url.hash = `json=${json.id},${exportedKey.k!}`;
       const urlString = url.toString();
-
       window.prompt(`🔒${t("alerts.uploadedSecurly")}`, urlString);
+    } else if (json.error_class === "RequestTooLargeError") {
+      window.alert(t("alerts.couldNotCreateShareableLinkTooBig"));
     } else {
       window.alert(t("alerts.couldNotCreateShareableLink"));
     }
   } catch (error) {
```
Comment (dwelle) [inline src/data/index.ts:227]:
```suggestion
  } catch (error) {
  	console.error(error);
```

We want to keep this so we log unexpected errors to our Sentry

File: src/locales/en.json [hunk 2]
```diff
@@ -106,6 +106,7 @@
   "alerts": {
     "clearReset": "This will clear the whole canvas. Are you sure?",
     "couldNotCreateShareableLink": "Couldn't create shareable link.",
+    "couldNotCreateShareableLinkTooBig": "Sorry we couldn't create your shareable link, your canvas is too big",
```
Comment (dwelle) [inline src/locales/en.json:109]:
```suggestion
    "couldNotCreateShareableLinkTooBig": "Couldn't create shareable link: the scene is too big",
```

Comment (dwelle) [reply src/locales/en.json:109]: "canvas" too big isn't really the problem here, it has more to do with payload size. Not that users will appreciate this distinction, but we usually use "scene" anyway.

**Review by dwelle — CHANGES_REQUESTED** [review]

**Comment by dwelle** [comment]
> P-S: In dev env the server isn't responding with error_class -> RequestTooLargeError

Works for me on dev too (which it should since on dev we still use the same server).

File: src/data/index.ts (hunk 1 above)
Comment (Pierre-Monier) [reply src/data/index.ts:227]: Husky doesn't accept console statement

**Comment by Pierre-Monier** [comment]
> > P-S: In dev env the server isn't responding with error_class -> RequestTooLargeError
> 
> Works for me on dev too (which it should since on dev we still use the same server).

I fork the excalidraw long time ago, I might not be up to date

**Review by dwelle — APPROVED** [review]

**Comment by dwelle** [comment]
Thanks for the PR!

**Pull request merged**